socs/agents/scpi_psu/agent.py

```diff
@@ -145,6 +226,26 @@ def set_current(self, session, params=None):
 
         return True, 'Set channel {} current to {}'.format(params['channel'], params['current'])
 
+    @ocs_agent.param('channel', type=int, choices=[1, 2, 3])
+    def get_output(self, session, params=None):
+        """get_output(channel)
+
+        **Task** - Check if channel ouput is enabled or disabled.
+
+        Parameters:
+            channel (int):
+        """
+        enabled = False
+        with self.lock.acquire_timeout(1) as acquired:
+            if acquired:
+                enabled = self.psu.get_output(params['channel'])
+            else:
+                return False, "Could not acquire lock."
+        if enabled:
+            return True, 'Channel {} output is currently enabled.'.format(params['channel'])
+        else:
+            return True, 'Channel {} output is currently disabled.'.format(params['channel'])
```

Comment:
This is alright, but only really shows up in the logs. Can you also add the information to the `session.data` like you did above? That allows control programs to retrieve the response, as it's available to the `OCSClient` object.

Reply:
I added it to the session data and made the appropriate changes to the docstring.

Reply:
Similar to the `session.data` structure for `get_current` and `get_voltage`, can we separate the channel from the enabled state key? Something like:
```
>>> response.session['data']
{'timestamp': 1723671503.4899583,
 'channel': 1,
 'state': 'enabled'}
```

or
```
>>> response.session['data']
{'timestamp': 1723671503.4899583,
 'channel': 1,
 'enabled': True}
```

Both for consistency with those other tasks and for ease of use on the client end. (It's simpler to check `resp.session['data']['state']` for any given channel rather than needing to build the `'channel_{number}'` string to get the key right.)

Reply:
Done

Reply:
This should be fixed now